enterprise/server/remote_execution/runner/runner.go

```diff
@@ -262,6 +265,26 @@ func (r *CommandRunner) Run(ctx context.Context) *interfaces.CommandResult {
 	return r.Container.Exec(ctx, command, nil, nil)
 }
 
+// syncWorkspace synchronizes the action workspace directory to the container.
+// It should be called after downloading inputs and before executing the action.
+//
+// This should only be called when using runner recycling, since the Run() API
+// is responsible for making the workspace available to the container, and
+// inputs have already been downloaded by the time Run() is called.
+//
+// This is also an NOP for configurations in which the runner shares the same
+// view of the workspace as the host, such as the workspace is mounted to a
+// container or made directly accessible to a bare runner.
+func (r *CommandRunner) syncWorkspace(ctx context.Context) error {
+	if r.VFS != nil {
+		return nil
+	}
+	if fc, ok := r.Container.Delegate.(*firecracker.FirecrackerContainer); ok {
+		return fc.SyncWorkspace(ctx)
```

Comment:
Would it work if we always called `SyncWorkspace` inside of firecracker `Exec` func? I think it would work if we changed `Create` to create a VM with a dummy workspace fs and then swap it out when `Exec` is called.

Reply:
Good call -- done